bundles/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/util/SlidingTimeWindow.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ambientweather.internal.util;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link SlidingTimeWindow} is responsible for managing a set of
+ * time-based values. This class is used to calculate trends over time.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+class SlidingTimeWindow<T> {
+    protected long period = 0;
+    protected final SortedMap<Long, T> storage = Collections.synchronizedSortedMap(new TreeMap<>());
```

Comment:
The field can be a "Map" as well if you don't use any of the SortedMap specific functions (and there are none afaik)

Reply:
I'm actually using the firstKey and lastKey methods of the SortedMap class.